native_extensions/emoji_selector/src/lib.rs

```diff
@@ -128,6 +136,35 @@ fn render_selector<'a>(
             emoji_selector.focus();
         "#;
 
+    let ch = use_coroutine(cx, |mut rx: UnboundedReceiver<Command>| {
+        to_owned![state];
+        async move {
+            let warp_cmd_tx = state.read().get_warp_ch();
```

Comment:
My friend, instead of save and read WARP_CH in state, I think it is better to use it directly here. What do you think?

And with this, we can remove code from state.

```suggestion
            let warp_cmd_tx = WARP_CMD_CH.tx.clone();
```

Reply:
@sdwoodbury did this part it was not working before his change so I suggest we leave this.

Reply:
@lgmarchi We tried that first and it didn't work.

Reply:
Ok, I will let it as resolved.

Reply:
@InfamousVague and @sdwoodbury, could we let a HACK comment? Because this situation is uncommon, and with HACK comment can avoid some questions in the future